packages/resources/src/RemixSite.ts

```diff
@@ -0,0 +1,1506 @@
+      origin,
+      allowedMethods: cloudfront.AllowedMethods.ALLOW_GET_HEAD_OPTIONS,
+      cachedMethods: cloudfront.CachedMethods.CACHE_GET_HEAD_OPTIONS,
+      compress: true,
+      cachePolicy: staticsCachePolicy,
+    };
+    for (const item of fs.readdirSync(publicPath)) {
+      if (item === "build") {
+        // This is the browser build, so it will have its own cache policy
+        staticsBehaviours["/build/*"] = {
+          ...staticBehaviourOptions,
+          cachePolicy: buildCachePolicy,
+        };
+      } else {
+        // This is a public asset, so it will use the public cache policy
+        const itemPath = path.join(publicPath, item);
+        staticsBehaviours[
+          fs.statSync(itemPath).isDirectory() ? `/${item}/*` : `/${item}`
+        ] = staticBehaviourOptions;
+      }
+    }
+
+    return staticsBehaviours;
+  }
+
+  private createCloudFrontBuildAssetsCachePolicy(): cloudfront.CachePolicy {
+    return new cloudfront.CachePolicy(
+      this,
+      "BuildCache",
+      RemixSite.buildCachePolicyProps
+    );
+  }
+
+  private createCloudFrontStaticsCachePolicy(): cloudfront.CachePolicy {
+    return new cloudfront.CachePolicy(
+      this,
+      "StaticsCache",
+      RemixSite.staticsCachePolicyProps
+    );
+  }
+
+  private createCloudFrontServerCachePolicy(): cloudfront.CachePolicy {
+    return new cloudfront.CachePolicy(
+      this,
+      "ServerCache",
+      RemixSite.serverCachePolicyProps
+    );
+  }
+
+  private createCloudFrontInvalidation(): CustomResource {
+    // Create a Lambda function that will be doing the invalidation
+    const invalidator = new lambda.Function(this, "CloudFrontInvalidator", {
+      code: lambda.Code.fromAsset(
+        path.join(__dirname, "../assets/BaseSite/custom-resource")
+      ),
+      layers: [this.awsCliLayer],
+      runtime: lambda.Runtime.PYTHON_3_7,
+      handler: "cf-invalidate.handler",
+      timeout: Duration.minutes(15),
+      memorySize: 1024,
+    });
+
+    // Grant permissions to invalidate CF Distribution
+    invalidator.addToRolePolicy(
+      new iam.PolicyStatement({
+        effect: iam.Effect.ALLOW,
+        actions: [
+          "cloudfront:GetInvalidation",
+          "cloudfront:CreateInvalidation",
+        ],
+        resources: ["*"],
+      })
+    );
+
+    const waitForInvalidation =
+      this.props.waitForInvalidation === false ? false : true;
+
+    return new CustomResource(this, "CloudFrontInvalidation", {
+      serviceToken: invalidator.functionArn,
+      resourceType: "Custom::SSTCloudFrontInvalidation",
+      properties: {
+        BuildId: this.isPlaceholder ? "live" : this.generateBuildId(),
+        DistributionId: this.cdk.distribution.distributionId,
+        // TODO: Ignore the browser build path as it may speed up invalidation
+        DistributionPaths: ["/*"],
+        WaitForInvalidation: waitForInvalidation,
+      },
+    });
```

Comment:
**thoughts:** We might want to do something a bit more clever here. Probably not for now, but in the case of server responses returning cache control headers we'll be invaliding their cache too right?

Reply:
On second thoughts that may be ok too as the underlying code would have changed. 🤔